lib/API/errors.cpp

```diff
@@ -138,4 +150,178 @@ llvm::Error emitDiagnostic(const OptDiagnosticCallback &onDiagnostic,
   return emitDiagnostic(onDiagnostic, diag, ec);
 }
 
+namespace {
```

Comment:
LLVM coding guidelines suggest we use `static` functions instead of anonymous namespaces. Only use anonymous namespaces for class definitions.

Reply:
Addressed

Reply:
As configured our clang-tidy does not agree. I am reverting to how I had the code before.